pkg/services/authz/mappers/rbac_mapper.go

```diff
@@ -1,41 +1,52 @@
 package mappers
 
-type VerbToAction map[string]string                            // e.g. "get" -> "read"
-type ResourceVerbToAction map[string]VerbToAction              // e.g. "dashboards" -> VerbToAction
-type GroupResourceVerbToAction map[string]ResourceVerbToAction // e.g. "dashboard.grafana.app" -> ResourceVerbToAction
+import (
+	"fmt"
 
-type ResourceToAttribute map[string]string                   // e.g. "dashboards" -> "uid"
-type GroupResourceToAttribute map[string]ResourceToAttribute // e.g. "dashboard.grafana.app" -> ResourceToAttribute
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+)
+
+const defaultAttribute = "uid"
+
+type VerbMapping map[string]string                           // e.g. "get" -> "read"
+type ResourceVerbMapping map[string]VerbMapping              // e.g. "dashboards" -> VerbToAction
+type GroupResourceVerbMapping map[string]ResourceVerbMapping // e.g. "dashboard.grafana.app" -> ResourceVerbToAction
+
+type ResourceAttributeMapping map[string]string                        // e.g. "dashboards" -> "uid"
+type GroupResourceAttributeMapping map[string]ResourceAttributeMapping // e.g. "dashboard.grafana.app" -> ResourceToAttribute
 
 type K8sRbacMapper struct {
-	DefaultActions   VerbToAction
-	DefaultAttribute string
-	Actions          GroupResourceVerbToAction
-	Attributes       GroupResourceToAttribute
+	GroupResourceVerbMapping      GroupResourceVerbMapping
+	GroupResourceAttributeMapping GroupResourceAttributeMapping
 }
 
 func NewK8sRbacMapper() *K8sRbacMapper {
+	defaultMapping := func(r string) VerbMapping {
+		return map[string]string{
+			utils.VerbGet:              fmt.Sprintf("%s:read", r),
+			utils.VerbList:             fmt.Sprintf("%s:read", r),
+			utils.VerbWatch:            fmt.Sprintf("%s:read", r),
+			utils.VerbCreate:           fmt.Sprintf("%s:create", r),
+			utils.VerbUpdate:           fmt.Sprintf("%s:write", r),
+			utils.VerbPatch:            fmt.Sprintf("%s:write", r),
+			utils.VerbDelete:           fmt.Sprintf("%s:delete", r),
+			utils.VerbDeleteCollection: fmt.Sprintf("%s:delete", r),
+			utils.VerbGetPermissions:   fmt.Sprintf("%s.permissions:read", r),
+			utils.VerbSetPermissions:   fmt.Sprintf("%s.permissions:write", r),
```

Comment:
This code seems to be duplicated with `pkg/services/accesscontrol/authorizer.go`. Can we re-use it somehow?

Reply:
We should most likely remove `pkg/services/accesscontrol/authorizer.go` and use a shared client for in-proc